integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java

```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+                    ControllerServiceReferencingComponentEntity::getId, ControllerServiceReferencingComponentEntity::getRevision));
+
+            //Stop the referencing processors and ensure they are in the stopped state
+            if (!processorRevisions.isEmpty()) {
+                String state = NifiProcessUtil.PROCESS_STATE.STOPPED.name();
+                log.info("Stopping all component references to controller service {} ", controllerService.getName());
+                UpdateControllerServiceReferenceRequestEntity stopComponentsRequest = new UpdateControllerServiceReferenceRequestEntity();
+                stopComponentsRequest.setState(state);
+                stopComponentsRequest.setId(controllerService.getId());
+                stopComponentsRequest.setReferencingComponentRevisions(processorRevisions);
+                updateReferences(id, stopComponentsRequest);
+                boolean updatedReferences = ensureComponentsAreOfState(id, "Processor", state, 5, 500, TimeUnit.MILLISECONDS);
+                if (!updatedReferences) {
+                    //error unable to change the state of the references. ... error
+                    throw new NifiClientRuntimeException("Unable to stop processor references to this controller service " + controllerService.getName() + " before making the update");
+                }
+            }
+
+            //disable any controller service references
+            if (!controllerServiceRevisions.isEmpty()) {
+                UpdateControllerServiceReferenceRequestEntity stopComponentsRequest = new UpdateControllerServiceReferenceRequestEntity();
+                stopComponentsRequest.setState("DISABLED");
+                stopComponentsRequest.setId(controllerService.getId());
+                stopComponentsRequest.setReferencingComponentRevisions(controllerServiceRevisions);
+                updateReferences(id, stopComponentsRequest);
+                boolean updatedReferences = ensureComponentsAreOfState(id, "ControllerService", "DISABLED", 5, 500, TimeUnit.MILLISECONDS);
+                if (!updatedReferences) {
+                    //error unable to change the state of the references. ... error
+                    throw new NifiClientRuntimeException(
+                        "Unable to disable other controller service references to this controller service " + controllerService.getName() + " before making the update");
+                }
+            }
+
+        }
+        //mark this controller service as disabled
+        log.info("Disabling the controller service  {} ", controllerService.getName());
+        //update the service and mark it disabled.  this will throw a RuntimeException if it is not successful
+        ControllerServiceDTO updatedService = updateStateByIdWithRetries(controllerService.getId(), "DISABLED", 5, 500, TimeUnit.MILLISECONDS);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "updatedService". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)